apps/openmw/mwrender/objects.cpp

```diff
@@ -197,9 +213,14 @@ void Objects::updatePtr(const MWWorld::Ptr &old, const MWWorld::Ptr &cur)
                 userDataContainer->setUserObject(i, new PtrHolder(cur));
         }
 
-    if (objectNode->getNumParents())
+    if (objectNode->getNumParents()>0)
+    {
         objectNode->getParent(0)->removeChild(objectNode);
-    cellnode->addChild(objectNode);
+    }
+
+    const float *f = cur.getRefData().getPosition().pos;
```

Comment:
Keep in mind getPosition().asVec3() is a thing, and it's not just here you can use it.

Reply:
i just used the original syntax...but if you think it's better, i have nothing against it